bin/gen_peptides.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+
+import sys
+import argparse
+import csv
+import pandas as pd
+from Bio import SeqIO
+from collections import Counter
+
+
+parser = argparse.ArgumentParser("Preprocessing of epitopes.")
+parser.add_argument('-i', '--input', metavar='FILE', type=str, help = 'TSV filename, containing paths to fasta files.')
+parser.add_argument('-o', '--output', metavar='FILE', type=argparse.FileType('w'), help='Output file: peptides.')
+parser.add_argument('-min', '--min_length', metavar='N', type=int, help='Min. length of peptides that will be generated.')
+parser.add_argument('-max', '--max_length', metavar='N', type=int, help='Max. length of peptides that will be generated.')
+args = parser.parse_args()
+
+
+# for each fasta file get proteins: Id and sequence
+def get_proteins(filename):
+    with open(filename, 'r') as fasta_file:
+        return [ (str(record.id), str(record.seq)) for record in SeqIO.parse(fasta_file, 'fasta') ]
+
+# for each file, get all proteins with id and sequence
+# not super efficient, but probably fine for now
+protid_protseq = pd.DataFrame(
+    [ (prot_id, prot_seq) for prot_id, prot_seq in get_proteins(args.input)  ],
+    columns = ['protein','sequence']
+    )
+
+
+####################
+# generate peptides
+def gen_peptides(prot_seq, k):
+    return [ prot_seq[i:(i+k)] for i in range(len(prot_seq)-k) ]
```

Comment:
same as above, we should use the FRED2 functionality to generate peptides from proteins in my opinion